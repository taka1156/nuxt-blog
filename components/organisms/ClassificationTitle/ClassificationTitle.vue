<template>
  <div>
    <div class="classification-title">
      <h1 class="classification-title__title">
        <!-- @slot タグもしくはカテゴリー名 -->
        <slot />
      </h1>
      <base-img :size="`lg`" :img-url="imgUrl" :img-alt="'ロゴ'" />
    </div>
  </div>
</template>

<script>
import BaseImg from '../../atoms/BaseImg/BaseImg';

export default {
  name: 'ClassificationTitle',
  components: {
    'base-img': BaseImg
  },
  props: {
    /**
     * 画像のURL
     */
    imgUrl: {
      type: String,
      default: '',
      require: true
    }
  }
};
</script>

<style scoped>
.classification-title {
  display: flex;
  justify-content: center;
}

.classification-title__title {
  padding: 5px;
  font-size: 1.25rem;
}

.classification-title__icon {
  width: 50px;
  height: 50px;
  padding: 5px;
}
</style>
