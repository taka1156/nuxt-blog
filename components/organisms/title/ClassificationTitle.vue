<template>
  <div>
    <div class="classification-title">
      <h1 class="classification-title__title">
        <!-- @slot タグもしくはカテゴリー名 -->
        <slot />
      </h1>
      <base-img :size="`lg`" :img-url="imgUrl" :img-alt="'ロゴ'" />
    </div>
  </div>
</template>

<script>
import BaseImg from '../../atoms/base/BaseImg';

export default {
  name: 'ClassficationTitle',
  components: {
    'base-img': BaseImg
  },
  props: {
    /**
     * 画像のURL
     */
    imgUrl: {
      type: String,
      default: '',
      require: true
    }
  }
};
</script>

<style scoped>
.classification-title {
  display: flex;
  justify-content: center;
}

.classification-title__title {
  padding: 5px;
  font-size: 1.25rem;
}

.classification-title__icon {
  padding: 5px;
  height: 50px;
  width: 50px;
}
</style>
