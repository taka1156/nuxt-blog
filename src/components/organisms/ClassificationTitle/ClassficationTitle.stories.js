import ClassificationTitle from './ClassificationTitle.vue';
import {
  dummyClassificationTag,
  dummyClassificationCategory
} from '__testdata__/testdata.js';

export default {
  title: 'Organisms/ClassificationTitle',
  component: ClassificationTitle
};

const Template1 = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { ClassificationTitle },
  template:
    '<classification-title v-bind="$props">ダミー : ダミータグ</classification-title>'
});

const Template2 = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { ClassificationTitle },
  template:
    '<classification-title v-bind="$props">ダミー : ダミーカテゴリー</classification-title>'
});

export const Tag = Template1.bind({});
Tag.args = {
  imgUrl: dummyClassificationTag.items[0].img.url
};

export const Category = Template2.bind({});
Category.args = {
  imgUrl: dummyClassificationCategory.items[0].img.url
};
