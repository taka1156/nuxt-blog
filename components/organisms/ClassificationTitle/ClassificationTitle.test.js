import { shallowMount } from '@vue/test-utils';
import ClassificationTitle from './ClassificationTitle.vue';
import {
  dummyClassificationCategory,
  dummyClassificationTag
} from '@/__testdata__/testdata.js';

describe('ClassificationTitle', () => {
  const classificationTitle = propsData =>
    shallowMount(ClassificationTitle, {
      propsData: {
        ...propsData
      }
    });

  it('ClassificationTitle初期値(category): imgUrl', () => {
    const wrapper = classificationTitle({
      imgUrl: dummyClassificationCategory.items[0].img.url
    });
    // imgUrl
    expect(wrapper.vm.$options.props.imgUrl.required).toBe(true);
    expect(wrapper.vm.imgUrl).toBe(dummyClassificationCategory.items[0].img.url);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('ClassificationTitle初期値(tag): imgUrl', () => {
    const wrapper = classificationTitle({
      imgUrl: dummyClassificationTag.items[0].img.url
    });
    // imgUrl
    expect(wrapper.vm.$options.props.imgUrl.required).toBe(true);
    expect(wrapper.vm.imgUrl).toBe(dummyClassificationTag.items[0].img.url);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
