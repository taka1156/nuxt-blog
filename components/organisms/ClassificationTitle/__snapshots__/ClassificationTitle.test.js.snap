// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassificationTitle ClassificationTitle初期値(category): imgUrl 1`] = `
"<div>
  <div class=\\"classification-title\\">
    <base-heading-stub hlv=\\"1\\"></base-heading-stub>
    <base-img-lazy-stub imgurl=\\"http://placehold.jp/150x150.png\\" imgalt=\\"ロゴ\\" size=\\"lg\\"></base-img-lazy-stub>
  </div>
  <div class=\\"classification-title__border\\"></div>
</div>"
`;

exports[`ClassificationTitle ClassificationTitle初期値(tag): imgUrl 1`] = `
"<div>
  <div class=\\"classification-title\\">
    <base-heading-stub hlv=\\"1\\"></base-heading-stub>
    <base-img-lazy-stub imgurl=\\"http://placehold.jp/150x150.png\\" imgalt=\\"ロゴ\\" size=\\"lg\\"></base-img-lazy-stub>
  </div>
  <div class=\\"classification-title__border\\"></div>
</div>"
`;
